osprepare/main.go

```diff
@@ -97,31 +159,38 @@ func collectPackages(dist distro, reqs PackageRequirements) []string {
 
 // InstallDeps installs all the dependencies defined in a component
 // specific PackageRequirements in order to enable running the component
-func InstallDeps(reqs PackageRequirements) {
+func InstallDeps(reqs PackageRequirements, logger OSPLog) {
+	if logger == nil {
+		logger = ospNullLogger{}
+	}
 	distro := getDistro()
 
 	if distro == nil {
-		fmt.Fprintf(os.Stderr, "Running on an unsupported distro\n")
+		logger.Errorf("Running on an unsupported distro")
 		if rel := getOSRelease(); rel != nil {
-			fmt.Fprintf(os.Stderr, "Unsupported distro: %s %s\n", rel.Name, rel.Version)
+			logger.Errorf("Unsupported distro: %s %s", rel.Name, rel.Version)
 		} else {
-			fmt.Fprintln(os.Stderr, "No os-release found on this host")
+			logger.Errorf("No os-release found on this host")
 		}
 		return
 	}
+	logger.Infof("OS Detected: %s", distro.getID())
+
 	// Nothing requested to install
 	if reqs == nil {
 		return
 	}
 	if reqPkgs := collectPackages(distro, reqs); reqPkgs != nil {
-		if distro.InstallPackages(reqPkgs) == false {
-			fmt.Fprintf(os.Stderr, "Failed to install: %s\n", strings.Join(reqPkgs, ", "))
+		logger.Infof("Missing packages detected: %v", reqPkgs)
+		if distro.InstallPackages(reqPkgs, logger) == false {
+			logger.Errorf("Failed to install: %s", strings.Join(reqPkgs, ", "))
```

Comment:
Okay, sorry, I do have a few more comments.  We probably want a return here otherwise we'll get 

Failed to install: ...
Missing packages installed

Also, I don't think Install packages is detecting errors correctly.  When I got an error from apt, I saw the following logs

```
I0916 14:38:04.155526    5546 main.go:177] OS Detected: ubuntu
I0916 14:38:04.155844    5546 main.go:184] Missing packages detected: [ceph-fs-common]
I0916 14:38:04.162392    5546 distro.go:242] E: Could not get lock /var/lib/dpkg/lock - open (11: Resource temporarily unavailable)
I0916 14:38:04.162654    5546 distro.go:242] E: Unable to lock the administration directory (/var/lib/dpkg/), is another process using it?
W0916 14:38:04.163153    5546 distro.go:246] Error running command: exit status 100
I0916 14:38:04.163394    5546 main.go:188] Missing packages installed.
```